clients/privacy-center/common/geolocation.ts

```diff
@@ -1,11 +1,15 @@
 import type { NextApiRequest } from "next";
 import { UserGeolocation } from "fides-js";
 
-// Regex to validate a location string, which must:
+// Regex to validate an ISO-3166 location string, which must:
 // 1) Start with a 2-3 character country code (e.g. "US")
-// 2) Optionally end with a 2-3 character region code (e.g. "CA")
+// 2) Optionally end with a 1-3 character region code (e.g. "CA")
 // 3) Separated by a dash (e.g. "US-CA")
-const VALID_ISO_3166_LOCATION_REGEX = /^\w{2,3}(-\w{2,3})?$/;
+const VALID_ISO_3166_LOCATION_REGEX = /^\w{2,3}(-\w{1,3})?$/;
```

Comment:
nit: _technically_ the `\w` metacharacter in regex would match some values that are invalid, e.g. `_` and also country codes that starts with a digit. 

This is how the regex was already built, so it doesn't make anything _worse_ as written. It's also of minimal risk IMO, but I just wanted to call it out.

The fix would be to replace your `\w` metacharacters with a set: `[A-Za-z]`

Reply:
Might also worth adding it to the unit tests 😄

Reply:
Er my last comment is a bit backwards, low reading comprehension there - but yes, I do think I should reject numeric country codes, that's just another subtle footgun!

Reply:
OK, updated! I think it's preferable to be defensive here and not overly permissive - the change to discard invalid  region headers is there as a safety hatch, but I think it's good to be strict about the `?geolocation=` param since that's a user-facing API